kvserver: add priority inversion metrics

wenyihu6 opened the pull request.

wenyihu6 committed:
callback

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/152696)
<!-- Reviewable:end -->

wenyihu6 committed:
add async

wenyihu6 committed:
kvserver: add enqueue metrics to base queue

Previously, observability into base queue enqueuing was limited to pending queue
length and processing rate. This commit adds enqueue specific metrics for the
replicate queue:
- queue.replicate.enqueue.success: replicas successfully enqueued
- queue.replicate.enqueue.failure: replicas that failed to enqueue
- queue.replicate.enqueue.skipped: replicas that were not enqueued
